doc/dev/release.rst

```diff
@@ -216,7 +214,7 @@ so:
         virtualenv build
         cd build
         source bin/activate
-        pip install -U setuptools==3.4.1 wheel
+        pip install wheel
```

Comment:
Adding a single data point here: I've used "the latest" (actually what ever conda gave me) version of setuptools and haven't run into problems yet.

I think it would be worth trying out a recent version for a while as a way to find out if we can move to that.